enterprise/server/remote_execution/runner/runner.go

```diff
@@ -471,11 +471,10 @@ func (p *Pool) add(ctx context.Context, r *CommandRunner) *labeledError {
 			"stats_failed",
 		}
 	}
-	// If memory usage stats are not implemented, fall back to the task size
-	// estimate.
+	// If memory usage stats are not implemented, fall back to the default task
+	// size estimate.
 	if stats.MemoryUsageBytes == 0 {
-		estimate := tasksize.Estimate(r.task)
```

Comment:
I'm having a hard time convincing myself that the newer code is more correct. What is the rationale for using the default task memory estimate as the basis of all task memory estimates?

If we think a workflow can use 8GB, then isn't it correct to say that we shouldn't allow more than 2 workflows in the pool? If the answer is that a workflow is not going to use 8GB then shouldn't we lower the workflow memory estimate so that we can fit more workflows in the pool? 

Sorry, not trying to be difficult. I know these memory estimates are a pain and perhaps I don't understand them well enough.

Reply:
I don't remember exactly, but I think the 8GB number accounts for the memory that bazel can be using while a build is in progress (e.g. local compile actions) while the effective 2.6GB "baseline" value is the memory used by bazel while it is sleeping between actions.

The name "default" is not great -- it's more of an "upperbound" estimate of the memory used by a paused, bare runner (for Docker, we can compute the memory of paused runners directly, and for Firecracker, the memory usage is zero since all resources are on disk, I think? Need to tweak that at some point, too)